test/metabase/api/search_test.clj

```diff
@@ -744,3 +763,73 @@
     (snowplow-test/with-fake-snowplow-collector
       (mt/user-http-request :crowberto :get 200 "search" :q "test" :archived true)
       (is (empty? (snowplow-test/pop-event-data-and-user-id!))))))
+
+;; ------------------------------------------------ Filter Tests ------------------------------------------------ ;;
+
+(deftest filter-by-creator-test
+  (let [search-term "Created by Filter"]
+    (with-search-items-in-root-collection search-term
+      (t2.with-temp/with-temp
+        [:model/User      {user-id :id}      {:first_name "Explorer" :last_name "Curious"}
+         :model/Card      {card-id :id}      {:name (format "%s Card 1" search-term) :creator_id user-id}
+         :model/Card      {card-id-2 :id}    {:name (format "%s Card 2" search-term) :creator_id user-id
+                                              :collection_id (:id (collection/user->personal-collection user-id))}
+         :model/Card      {card-id-3 :id}    {:name (format "%s Card 3" search-term) :creator_id user-id :archived true}
+         :model/Card      {model-id :id}     {:name (format "%s Dataset 1" search-term) :dataset true :creator_id user-id}
+         :model/Dashboard {dashboard-id :id} {:name (format "%s Dashboard 1" search-term) :creator_id user-id}
+         :model/Action    {action-id :id}    {:name (format "%s Action 1" search-term) :model_id model-id :creator_id user-id :type :http}]
+
+        (testing "sanity check that without search by created_by we have more results than if a filter is provided"
+          (is (> (:total (mt/user-http-request :crowberto :get 200 "search" :q search-term))
+                 5)))
+
+        (testing "Able to filter by creator"
+          (let [resp (mt/user-http-request :crowberto :get 200 "search" :q search-term :created_by user-id)]
+
+            (testing "only a subset of models are applicable"
+              (is (= #{"card" "dataset" "dashboard" "action"} (set (:available_models resp)))))
+
+            (testing "results contains only entities with the specified creator"
+              (is (= [[dashboard-id "dashboard" "Created by Filter Dashboard 1"]
+                      [card-id      "card"      "Created by Filter Card 1"]
+                      [card-id-2    "card"      "Created by Filter Card 2"]
+                      [model-id     "dataset"   "Created by Filter Dataset 1"]
+                      [action-id    "action"    "Created by Filter Action 1"]]
+                     (->> (:data resp)
+                          sorted-results
+                          (map (juxt :id :model :name))))))))
```

Comment:
The order of the search results are underdetermined here, since there are more than one result with the same model and therefore scoring. So this test can flake, as it is doing here 
https://github.com/metabase/metabase/actions/runs/5654663460/job/15318203858?pr=32469#step:4:875

One solution is to drop trying to test the order of results here, and only care about the set of results in this test.